addons/calendar/models/calendar_event.py

```diff
@@ -989,11 +989,7 @@ def action_unlink_event(self, attendee_id=None, recurrence=False):
             _logger.warning('Template "calendar.calendar_template_delete_event" was not found. Cannot send delete notifications.')
             return {}
 
-        lang = self.env.context.get('lang')
-        if template.lang:
-            # This method ensures that the template is translated according
-            # to the user's or record's language settings.
-            lang = template._render_lang(self.ids)[self.id]
+        if lang := template._render_lang(self.ids)[self.id]:
```

Comment:
```suggestion
        if lang := self.ids and template._render_lang(self.ids)[self.id]:
```
self.ids should never be empty but I prefer to take into account this case instead of having a traceback. What do you think ? The other solution is 
```suggestion
        if lang := template._render_lang(self.ids).get(self.id):
```
You could face the issue if the metnod was called with new NewId instance. self.ids is empty in that case and self.id is the NewId value.